# End2End Controller Specific Configuration
# This file contains parameters specific to the End2End controller

controller_executor:
  ros__parameters:
    
    # Controller type selection
    controller_type: "end2end"
    
    model_path: "example_controllers/end2end/end2end_policy/aow_end2end_policy_3.pt"
    
    trajectory_file: "utils/trajectory_generation/example_trajectory_csv/rsl_path_scale5.0_rounds1.csv"

    # Joint position offset flags, linked to Isaaclab's action term 
    use_default_joint_positions_offset_observation: false
    use_default_joint_positions_offset_action: true

    # End2end controller specific parameters
    num_waypoints: 10           # Number of waypoints to extract for trajectory following
    search_range: 120           # Forward search window for finding closest waypoint  
    fixed_interval_m: 0.30      # Fixed interval distance in meters between waypoints

